File: backend/student_management/src/main/java/com/example/student_management/entity/Department.java
package com.example.student_management.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
@Entity
@Table(name = "department")
public class Department {
    @Id
    @Column(name = "id", length = 10)
    private String id;

    @Column(nullable = false)
    private String name;

    @Column(unique = true, nullable = false)
    private String code;

    private String headLecturerId;

    @Column(columnDefinition = "TEXT")
    private String description;

    @OneToMany(mappedBy = "department")
    private List<User> users;

    @OneToMany(mappedBy = "department")
    private List<SubjectEntity> subjects;

    @OneToMany(mappedBy = "department")
    private List<SchoolClass> schoolClasses;

}

File: mobile/student_management_flutter/lib/models/department.dart
class Department {
  final String id;
  final String name;
  final String code;
  final String? description;

  Department({
    required this.id,
    required this.name,
    required this.code,
    this.description,
  });

  factory Department.fromJson(Map<String, dynamic> json) {
    return Department(
      id: json['id'] as String,
      name: json['name'] as String,
      code: json['code'] as String,
      description: json['description'] as String?,
    );
  }
}

